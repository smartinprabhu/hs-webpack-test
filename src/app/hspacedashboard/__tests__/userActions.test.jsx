import React from 'react';
import { useSelector } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent, cleanup } from '@testing-library/react';
import UserActions from '../userActions';

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

jest.mock('react-redux', () => ({
  useSelector: jest.fn((fn) => fn()),
  useDispatch: () => jest.fn(),
}));

jest.mock('../../data/userActions.json', () => ({
  data: [{
    id: 1,
    name: 'My Bookings',
    image: 'src/app/images/myBookingsBlue.ico',
    route: '/bookings',
  },
  {
    id: 2,
    name: 'Raise A Ticket',
    image: 'src/app/images/covidIncidentBlue.ico',
    route: '/raise-a-ticket',
  },
  {
    id: 3,
    name: 'My Guests',
    image: 'src/app/images/myguests.ico',
    route: '/myguests',
  }],
}));

const userInfo = {
  allowed_companies: [
    {
      country: {
        id: 104,
        name: 'India',
      },
      id: 3,
      locality: 'Bengaluru',
      name: 'Divyasree Technopark..',
      postal_code: '560037',
      region: {
        id: 593,
        name: 'Karnataka (IN)',
      },
      street_address: 'DTP Entry Gate, EPIP Zone',
    },
  ],
  company: {
    address: 'Divyasree Technopark..\nDTP Entry Gate, EPIP Zone\nBrookefield\nBengaluru 560037\nKarnataka KA\nIndia',
    id: 3,
    name: 'Divyasree Technopark..',
    timezone: 'Asia/Kolkata',
  },
  id: 9,
  image_url: 'https://demo.helixsense.com/web/image/message.queue.mqtt/1/avatar_image/1.jpeg',
  name: 'Technopark',
  email: {
    email: 'example@gmail.com',
  },
};

afterEach(cleanup);

describe('useractions testing', () => {
  // snapshot testing

  it('snapshot testing', () => {
    useSelector.mockImplementation((fn) => fn({
      user: {
        userInfo,
      },
      config: {
        configObj: {
          data: {},
        },
      },
    }));
    const { asFragment } = render(<UserActions />);
    expect(asFragment()).toMatchSnapshot();
  });

  // dom testing

  it('dom testing', () => {
    useSelector.mockImplementation((fn) => fn({
      user: {
        userInfo,
      },
      config: {
        configObj: {
          data: {},
        },
      },
    }));
    const { container, getByText } = render(<UserActions />);
    expect(getByText('My Actions')).toBeInTheDocument();
    expect(getByText('My Bookings')).toBeInTheDocument();
    fireEvent.click(getByText('My Bookings'));
    fireEvent.click(getByText('My Guests'));
    expect(mockHistoryPush).toHaveBeenCalledWith('/myguests');
    expect(container).toBeInTheDocument();
  });
});
