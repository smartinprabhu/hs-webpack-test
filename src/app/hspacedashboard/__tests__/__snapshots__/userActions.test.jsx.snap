// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useractions testing snapshot testing 1`] = `
<DocumentFragment>
  <div
    class="bg-azure rounded m-0 p-3 dashboardTileHeight row"
  >
    <div
      class="col-sm-10 col-md-10 col-lg-10"
    >
      <h2>
        My Actions
      </h2>
      <div
        class="mt-3 card"
      >
        <div
          class="p-3 card-body"
        >
          <div
            class="row"
          >
            <div
              class="col-3 col-sm-2 col-md-2 col-lg-2"
            >
              <img
                alt="actions"
                height="45"
                width="45"
              />
            </div>
            <div
              class="pl-0 col-7 col-sm-8 col-md-8 col-lg-8"
            >
              <h3
                class="mt-2 ml-2 cursor-pointer font-weight-700 pr-0"
              >
                My Bookings
              </h3>
            </div>
            <div
              class="col-2 col-sm-2 col-md-2 col-lg-2"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root float-right mt-2 font-size-30px"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mt-3 card"
      >
        <div
          class="p-3 card-body"
        >
          <div
            class="row"
          >
            <div
              class="col-3 col-sm-2 col-md-2 col-lg-2"
            >
              <img
                alt="actions"
                height="45"
                src="[object Object]"
                width="45"
              />
            </div>
            <div
              class="pl-0 col-7 col-sm-8 col-md-8 col-lg-8"
            >
              <h3
                class="mt-2 ml-2 cursor-pointer font-weight-700 pr-0"
              >
                My Guests
              </h3>
            </div>
            <div
              class="col-2 col-sm-2 col-md-2 col-lg-2"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root float-right mt-2 font-size-30px"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mt-10 ml-11 mb-2 p-0 float-right col-sm-12"
    >
      <img
        alt="Helixsense Portal"
        class="mt-auto w-inherit"
        height="140"
        src="[object Object]"
      />
    </div>
  </div>
</DocumentFragment>
`;
